tasks/change-granule-collection-s3/tests/test-index.js

```diff
@@ -0,0 +1,1000 @@
+          fileName: 'MOD11A1.A2017200.h19v04.006.2017201090724_1.jpg',
+          bucket: t.context.privateBucket,
+          type: 'browse',
+        },
+        {
+          key: 'file-staging/subdir/MOD11A1.A2017200.h19v04.006.2017201090724_2.jpg',
+          fileName: 'MOD11A1.A2017200.h19v04.006.2017201090724_2.jpg',
+          bucket: t.context.publicBucket,
+          type: 'browse',
+        },
+        {
+          key: 'file-staging/subdir/MOD11A1.A2017200.h19v04.006.2017201090724.ummg.cmr.json',
+          fileName: 'MOD11A1.A2017200.h19v04.006.2017201090724.ummg.cmr.json',
+          bucket: t.context.protectedBucket,
+          type: 'metadata',
+        },
+      ],
+    },
+    bad_granule: {
+      status: 'completed',
+      collectionId: 'MOD11A1___006',
+      granuleId: 'MOD11A1.A2017200.h19v04.006.2017201090724',
+      files: [
+        {
+          key: 'file-staging/subdir/MOD11A1.A2017200.h19v04.006.2017201090724.hdf',
+          fileName: 'MOD11A1.A2017200.h19v04.006.2017201090724.hdf',
+          type: 'data',
+        },
+        {
+          key: 'file-staging/subdir/MOD11A1.A2017200.h19v04.006.2017201090724_1.jpg',
+          fileName: 'MOD11A1.A2017200.h19v04.006.2017201090724_1.jpg',
+          bucket: t.context.privateBucket,
+          type: 'browse',
+        },
+        {
+          key: 'file-staging/subdir/MOD11A1.A2017200.h19v04.006.2017201090724_2.jpg',
+          fileName: 'MOD11A1.A2017200.h19v04.006.2017201090724_2.jpg',
+          bucket: t.context.publicBucket,
+          type: 'browse',
+        },
+        {
+          key: 'file-staging/subdir/MOD11A1.A2017200.h19v04.006.2017201090724.cmr.xml',
+          fileName: 'MOD11A1.A2017200.h19v04.006.2017201090724.cmr.xml',
+          bucket: t.context.protectedBucket,
+          type: 'metadata',
+        },
+      ],
+    },
+  }[granuleId];
+}
+
+function granulesToFileURIs(granuleIds, t) {
+  const granules = granuleIds.map((granuleId) => dummyGetGranule(granuleId, t));
+  const files = granules.reduce((arr, g) => arr.concat(g.files), []);
+  return files.map((file) => buildS3Uri(file.bucket, file.key));
+}
+
+function buildPayload(t, collection) {
+  const newPayload = t.context.payload;
+  newPayload.config.targetCollection = collection;
+  newPayload.config.bucket = t.context.stagingBucket;
+  newPayload.config.buckets.internal.name = t.context.stagingBucket;
+  newPayload.config.buckets.public.name = t.context.publicBucket;
+  newPayload.config.buckets.private.name = t.context.privateBucket;
+  newPayload.config.buckets.protected.name = t.context.protectedBucket;
+  return newPayload;
+}
+
+test.beforeEach(async (t) => {
+  const topicName = randomString();
+  const { TopicArn } = await createSnsTopic(topicName);
+  process.env.granule_sns_topic_arn = TopicArn;
+  const testDbName = `change-granule-collection-s3/change-collections-s3${cryptoRandomString({ length: 10 })}`;
+  changeGranuleCollectionS3 = proxyquire(
```

Comment:
If at all possible, we should avoid proxyquire.     Is there another way to do this?

Reply:
I'd suggest we use a test context object on the event, or as another param on the method that allows us to pass in a mocked object/sinon object here, particularly given getGranule is a first level method.

Reply:
Thanks for sorting!